driver/pace/driver/diagnostics.py

```diff
@@ -34,6 +36,35 @@ def cleanup(self):
         ...
 
 
+@dataclasses.dataclass
+class ZSelect:
+    level: int
+    names: List[str]
+
+    def slice_data(self, state: DycoreState):
+        output = {}
+        for name in self.names:
+            if name not in state.__dict__.keys():
+                raise ValueError(f"Invalid state variable {name} for level select")
+            assert len(getattr(state, name).dims) > 2
+            if getattr(state, name).dims[2] != (
+                pace.util.Z_DIM or pace.util.Z_INTERFACE_DIM
+            ):
+                raise ValueError(
+                    f"z_select only works for state variables with a z dimension. \
```

Comment:
we don't actually have a field that triggers this, right?

Reply:
Correct, while writing test I realized this is not possible based on the current DycoreState.

Reply:
Added test